jenkins/bootstrap.py

```diff
@@ -320,14 +324,18 @@ def copy_file(self, dest, orig):
         cmd = [self.gsutil, '-q', 'cp', '-Z', orig, dest]
         self.call(cmd)
 
-    def upload_text(self, path, txt, cached=True):
+    def upload_text(self, path, txt, meta_links={}, cached=True):
         """Copy the text to path, optionally disabling caching."""
         headers = ['-h', 'Content-Type:text/plain']
         if not cached:
             headers += ['-h', 'Cache-Control:private, max-age=0, no-transform']
         cmd = [self.gsutil, '-q'] + headers + ['cp', '-', path]
         self.call(cmd, stdin=txt)
 
+        for meta, link in meta_links.iteritems():
+            cmd = [self.gsutil, 'setmeta', '-h', '\'%s: %s\'' % (meta, link), path]
```

Comment:
gsutil cp writes metadata already, (Cache-Control etc are metadata), setmeta just modifies the metadata on an existing object.